### ISSUE TITLE
garbled code when throw Error with non-ASCII message in Deno.test

### ISSUE THREAD

**Issue by sdweiyu** [denoland/deno#9966]
```typescript
Deno.test({
  name: "test",
  fn(): void {
    throw Error("抛出异常");
  },
});
```
output:
```
running 1 tests
test test ... FAILED (9ms)

failures:

test
Error: 鎶涘嚭寮傚父
    at fn (file:///C:/Users/....
```

or

```typescript
Deno.test({
  name: "test",
  fn(): void {
    asserts.assertThrows(
      (): void => {
        throw Error("抛出异常");
      },
      Error,
      `抛出异常1`,
    );
  },
});
```
output:
```
running 1 tests
test test ... FAILED (15ms)

failures:

test
AssertionError: Expected error message to include "鎶涘嚭寮傚父1", but got "鎶涘嚭寮傚父".
    at Module.assertThrows (file:///C:/Users
```

**Comment by littledivy** [comment]
@sdweiyu I am unable to reproduce this in Deno 1.8.3

```typescript
Deno.test({
  name: "test",
  fn: () => {
    throw Error("抛出异常");
  },
});
```

```bash
test test ... FAILED (7ms)

failures:

test
Error: 抛出异常
```
The output looks as expected. Can you try upgrading to latest Deno release?

**Comment by kitsonk** [comment]
Also, please provide the OS/terminal you are using if the problem persists.

**Comment by pedropaulosuzuki** [comment]
Probably related to #8239.

**Comment by pedropaulosuzuki** [comment]
> ```ts
> Deno.test({
>   name: "test",
>   fn(): void {
>     throw Error("抛出异常");
>   },
> });
> ```
> 
> output:
> 
> ```
> running 1 tests
> test test ... FAILED (9ms)
> 
> failures:
> 
> test
> Error: 鎶涘嚭寮傚父
>     at fn (file:///C:/Users/....
> ```
> 
> or
> 
> ```ts
> Deno.test({
>   name: "test",
>   fn(): void {
>     asserts.assertThrows(
>       (): void => {
>         throw Error("抛出异常");
>       },
>       Error,
>       `抛出异常1`,
>     );
>   },
> });
> ```
> 
> output:
> 
> ```
> running 1 tests
> test test ... FAILED (15ms)
> 
> failures:
> 
> test
> AssertionError: Expected error message to include "鎶涘嚭寮傚父1", but got "鎶涘嚭寮傚父".
>     at Module.assertThrows (file:///C:/Users
> ```

Actually I think you are getting this error because you added a 1 after the expected value in your test.

**Comment by sdweiyu** [comment]
@kitsonk  @littledivy  Sorry, I’m late.
OS: Win10 10.0.19042.906
terminal: I tried PowerShell and cmd.  Codepage 936. 
I tried deno 1.2.1, 1.5.4 1.8.2,1.8.3.

**Comment by kitsonk** [comment]
And:

```
console.log("抛出异常");
```

Works as expected?

**Comment by sdweiyu** [comment]
> And:
> 
> ```ts
> console.log("抛出异常");
> ```
> 
> Works as expected?

yes

**Comment by caspervonb** [comment]
Don't have a windows machine to test this on but based on what test does I believe this should be enough to reproduce.

```js
Deno.stdout.write("鎶涘嚭寮傚");
Deno.stdout.write("\n");
```

**Comment by caspervonb** [comment]
Tried this on Windows, and I always get gabled output with stdout.write.

```
await Deno.stdout.write(new TextEncoder().encode("抛出异常\n"));
µè¢σç║σ╝éσ╕╕
```

console.log works correctly.

**Comment by F3n67u** [comment]
There is no activity here for the last 2 months, let's close this issue for now and re-open if needed.

**Comment by pedropaulosuzuki** [comment]
> There is no activity here for the last 2 months, let's close this issue for now and re-open if needed?

I think keeping it open until it is solved is a better practice, so that we get a better view on the state of the art in Deno. But it would be nice to confirm if the issue still persists, as it might have been solved already in some random commit.

**Comment by ayame113** [comment]
I have confirmed that the issue has been fixed in windows.

test.ts:
```js
Deno.test({
  name: "test",
  fn(): void {
    throw Error("抛出异常");
  },
});
```

output:
```
running 1 test from file:///C:/Users/azusa/tmp/test.ts
test test ... FAILED (18ms)

failures:

test
Error: 抛出异常
    at fn (file:///C:/Users/azusa/tmp/test.ts:4:11)
    at asyncOpSanitizer (deno:runtime/js/40_testing.js:35:15)
    at resourceSanitizer (deno:runtime/js/40_testing.js:72:13)
    at exitSanitizer (deno:runtime/js/40_testing.js:99:15)
    at runTest (deno:runtime/js/40_testing.js:215:13)
    at Object.runTests (deno:runtime/js/40_testing.js:283:28)
    at file:///C:/Users/azusa/tmp/69e04224-12ed-4f49-9210-ed13a39b18ea$deno$test.js:4:33

failures:

        test

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out (61ms)

error: Test failed
```

**Comment by caspervonb** [comment]
> I have confirmed that the issue has been fixed in windows.
> 
> test.ts:
> 
> ```js
> Deno.test({
>   name: "test",
>   fn(): void {
>     throw Error("抛出异常");
>   },
> });
> ```
> 
> output:
> 
> ```
> running 1 test from file:///C:/Users/azusa/tmp/test.ts
> test test ... FAILED (18ms)
> 
> failures:
> 
> test
> Error: 抛出异常
>     at fn (file:///C:/Users/azusa/tmp/test.ts:4:11)
>     at asyncOpSanitizer (deno:runtime/js/40_testing.js:35:15)
>     at resourceSanitizer (deno:runtime/js/40_testing.js:72:13)
>     at exitSanitizer (deno:runtime/js/40_testing.js:99:15)
>     at runTest (deno:runtime/js/40_testing.js:215:13)
>     at Object.runTests (deno:runtime/js/40_testing.js:283:28)
>     at file:///C:/Users/azusa/tmp/69e04224-12ed-4f49-9210-ed13a39b18ea$deno$test.js:4:33
> 
> failures:
> 
>         test
> 
> test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out (61ms)
> 
> error: Test failed
> ```

That's for a different reason, reportering no longer happens in JavaScript.

> Tried this on Windows, and I always get gabled output with stdout.write.
> 
> ```
> await Deno.stdout.write(new TextEncoder().encode("抛出异常\n"));
> µè¢σç║σ╝éσ╕╕
> ```
> 
> console.log works correctly.



> I have confirmed that the issue has been fixed in windows.
> 
> test.ts:
> 
> ```js
> Deno.test({
>   name: "test",
>   fn(): void {
>     throw Error("抛出异常");
>   },
> });
> ```
> 
> output:
> 
> ```
> running 1 test from file:///C:/Users/azusa/tmp/test.ts
> test test ... FAILED (18ms)
> 
> failures:
> 
> test
> Error: 抛出异常
>     at fn (file:///C:/Users/azusa/tmp/test.ts:4:11)
>     at asyncOpSanitizer (deno:runtime/js/40_testing.js:35:15)
>     at resourceSanitizer (deno:runtime/js/40_testing.js:72:13)
>     at exitSanitizer (deno:runtime/js/40_testing.js:99:15)
>     at runTest (deno:runtime/js/40_testing.js:215:13)
>     at Object.runTests (deno:runtime/js/40_testing.js:283:28)
>     at file:///C:/Users/azusa/tmp/69e04224-12ed-4f49-9210-ed13a39b18ea$deno$test.js:4:33
> 
> failures:
> 
>         test
> 
> test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out (61ms)
> 
> error: Test failed
> ```

Unfortunately, that's just because the test reporter is no longer written in JS. The minimal repro above should still be failing.

**Comment by ayame113** [comment]
Sure, the Deno.stdout issue in the comments still seems to be occurring.👀
```
Windows 10.0.19042.1110
Deno 1.12.2
```

```shell
C:\Users\azusa>deno
Deno 1.12.2
exit using ctrl+d or close()
> await Deno.stdout.write(new TextEncoder().encode("抛出异常\n"));
謚帛・蠑ょｸｸ
13
> await Deno.stdout.write(new TextEncoder().encode("こんにちは\n"));
縺薙ｓ縺ｫ縺｡縺ｯ
16
>
```

**Comment by sdweiyu** [comment]
```
Deno.test({
  name: "test",
  fn(): void {
    throw Error("抛出异常");
  },
});
```

This has been fixed from the release 1.10.1
---------------------------------
```
C:\Users\azusa>deno
Deno 1.12.2
exit using ctrl+d or close()
> await Deno.stdout.write(new TextEncoder().encode("抛出异常\n"));
謚帛・蠑ょｸｸ
13
> await Deno.stdout.write(new TextEncoder().encode("こんにちは\n"));
縺薙ｓ縺ｫ縺｡縺ｯ
16
>
```

This has been fixed from the release 1.21.3
--------------
